.circleci/config.yml

```diff
@@ -424,25 +424,113 @@ jobs:
             terraform output -json >> terraform_output.json
             export LOAD_BALANCER=`cat terraform_output.json | jq -r '.load_balancer_raw_dns' |jq -r '.value'`
             echo $LOAD_BALANCER
-            curl $LOAD_BALANCER
+            curl "$LOAD_BALANCER/hello"
             echo "1"
       - run:
           name: "Sleep 120 seconds"
           command: |
             sleep 120
       - run:
-          name: "Destroy Terraform Generated Http Service"
+          name: "Destroy Terraform Generated Hello Opta Service"
           command: |
             cd $HOME/generated-tf-files-aws
-            terraform plan -compact-warnings -lock=false -input=false -out=tf.plan -target=module.app -destroy
+            terraform plan -compact-warnings -lock=false -input=false -out=tf.plan -target=module.hello -destroy
             terraform apply -compact-warnings -auto-approve tf.plan
       - persist_to_workspace:
           root: ~/
           paths:
             - generated-tf-files-aws/tfstate/awsenv-tg.tfstate
       - store_artifacts:
           path: ../generated-tf-files-aws/tfstate/awsenv-tg.tfstate
-          destination: tfstate-after-apply-svc-cnd.tfstate
+          destination: tfstate-after-apply-hello-cnd.tfstate
+
+  aws-tg-test-other-modules-plan:
+    executor: ubuntu-machine
+    steps:
+      - checkout-opta
+      - install-opta-dependencies
+      - attach_workspace:
+          at: ~/
+      - attach_workspace:
+          at: ./
+      - run:
+          name: "Generate Terraform Language Files for DocDB"
+          command: |
+            ./dist/opta/opta generate-terraform \
+            --config ./.circleci/ci-tests/terraform-generated-aws/module-docdb.yaml \
+            --directory $HOME/generated-tf-files-aws \
+            --auto-approve
+      - run:
+          name: "Plan Terraform Generated for DocDB"
+          command: |
+            cd $HOME/generated-tf-files-aws
+            terraform init
+            terraform plan -compact-warnings -lock=false -input=false -out=tf.plan -target=module.tgdocdb
```

Comment:
anyway we can avoid repeating this code?

this post is interesting, I wonder if we could something similar:
https://discuss.circleci.com/t/circleci-orb-with-array-parameters-and-for-loop/36964/2

```
          command: |
            printf "Validating terraform code\n"
            for i in <<parameters.modules>>; do
               # find  opta file in module directory.
               # if not found, generate a default opta file to use with the module to test
               # run opta generate-terraform
               # run terraform init, plan
            done
```

Reply:
How about we do something like this?

```yaml
version: 2.1
commands: # a reusable command with parameters
  opta-module-with-terraform-generate:
    parameters:
      module:
        default: ""
        type: string
    steps:
      - run:
          name: Generate Terraform Language Files for << parameters.module >>
          command: |
            ./dist/opta/opta generate-terraform \
            --config ./.circleci/ci-tests/terraform-generated-aws/module-docdb.yaml \
            --directory $HOME/generated-tf-files-aws \
            --auto-approve
      - run:
          name: Terraform Plan for << parameter.module >>
          commamd: |
            cd $HOME/generated-tf-files-aws
            terraform init
            terraform plan -compact-warnings -lock=false -input=false -out=tf.plan -target=module.<< parameter.module >>
jobs:
  test-terraform-module:
    executor: ubuntu-machine
    steps:
      - opta-module-with-terraform-generate:
          module: "docdb"
      - opta-module-with-terraform-generate:
          module: "s3"
workflows:
  my-workflow:
    jobs:
      - test-terraform-module
```

Reply:
@nsarupr sorry for responding only now, it looks amazing, great case of reuse vs duplicate 💯